main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -2007,6 +2018,17 @@ public View getView(final int position, final View convertView, @NonNull final V
                 final CachedetailWaypointsHeaderBinding headerBinding = CachedetailWaypointsHeaderBinding.inflate(getLayoutInflater(), v, false);
                 v.addHeaderView(headerBinding.getRoot());
 
+                final CachedetailWaypointsCheckboxBinding checkboxBinding = CachedetailWaypointsCheckboxBinding.inflate(getLayoutInflater(), v, false);
+                v.addHeaderView(checkboxBinding.getRoot());
+
+                checkboxBinding.hideVisitedWaypoints.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                    @Override
+                    public void onCheckedChanged(final CompoundButton buttonView, final boolean isChecked) {
+                        hideVisitedWaypoints = !hideVisitedWaypoints;
```

Comment:
should be hideVisitedWaypoints = isChecked

Reply:
made correction in next commit